## IMDb 1,000 popular movies from 2006-2016 Analysis

### This document will analyze 1,000 popular movies taken from the IMDb website that looks at data points such as title, genre, director, revenue, etc. This analysis aims to see among popular movies if there are correlations between their success and exciting results, such as the average runtime of these 1,000 movies and their average revenue.

We will be importing Pandas so that we can analyze our data. I have gotten the metadata for this analysis from [Kaggle](https://www.kaggle.com/datasets/PromptCloudHQ/imdb-data).

Below is where we import Pandas and where our data index happens. This is going to put our 1,000 movies into columns and rows. We also have **index_col="Title"**, which will be used later on to type in a movie title and get the data such as the genre, actors, director, rating, etc.


In [6]:
import pandas as pd
data = pd.read_csv('IMDB-Movie-Data.csv')
data_indexed = pd.read_csv('IMDB-Movie-Data.csv', index_col="Title")

Using **data.head()** will print out the first five rows of our data.

In [7]:
data.head()

Rank                    Title                     Genre  \
0     1  Guardians of the Galaxy   Action,Adventure,Sci-Fi   
1     2               Prometheus  Adventure,Mystery,Sci-Fi   
2     3                    Split           Horror,Thriller   
3     4                     Sing   Animation,Comedy,Family   
4     5            Suicide Squad  Action,Adventure,Fantasy   

                                         Description              Director  \
0  A group of intergalactic criminals are forced ...            James Gunn   
1  Following clues to the origin of mankind, a te...          Ridley Scott   
2  Three girls are kidnapped by a man with a diag...    M. Night Shyamalan   
3  In a city of humanoid animals, a hustling thea...  Christophe Lourdelet   
4  A secret government agency recruits some of th...            David Ayer   

                                              Actors  Year  Runtime (Minutes)  \
0  Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014                121   
1  Noomi Rapace, Logan Marshall-Green, Michael Fa...  2012                124   
2  James McAvoy, Anya Taylor-Joy, Haley Lu Richar...  2016                117   
3  Matthew McConaughey,Reese Witherspoon, Seth Ma...  2016                108   
4  Will Smith, Jared Leto, Margot Robbie, Viola D...  2016                123   

   Rating   Votes  Revenue (Millions)  Metascore  
0     8.1  757074              333.13       76.0  
1     7.0  485820              126.46       65.0  
2     7.3  157606              138.12       62.0  
3     7.2   60545              270.32       59.0  
4     6.2  393727              325.02       40.0

With **data.info()**, we can see how many entries there are in our metadata which is 1,000. We can also see that there are twelve columns, which is the categories for each movie, as seen below.

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Rank                1000 non-null   int64  
 1   Title               1000 non-null   object 
 2   Genre               1000 non-null   object 
 3   Description         1000 non-null   object 
 4   Director            1000 non-null   object 
 5   Actors              1000 non-null   object 
 6   Year                1000 non-null   int64  
 7   Runtime (Minutes)   1000 non-null   int64  
 8   Rating              1000 non-null   float64
 9   Votes               1000 non-null   int64  
 10  Revenue (Millions)  872 non-null    float64
 11  Metascore           936 non-null    float64
dtypes: float64(3), int64(4), object(5)
memory usage: 93.9+ KB


**data.shape** tells us how many rows and columns there are. There are 1,000 rows and 12 columns.

In [9]:
data.shape

(1000, 12)

When using **data.describe()**, this is where much of the meat is regarding the data for these movies. We can see the mean runtime of the 1,000 movies, roughly 113 minutes, along with the standard deviation and percentiles in the runtime, rating, revenue, and Metascore of the movies. We also see the average rating, roughly 6.7/10, and the minimum and maximum revenue, which, to clarify, is the domestic revenue, not the worldwide revenue. We can see the movie from the list with the highest revenue was roughly 937 million dollars, and the movie it says with the lowest revenue was zero dollars. When I looked online, this wasn't correct since the movie from the list that it says made zero dollars was a movie called **A Kind of Murder**. This movie actually made $2,915 domestically. Because of this, the movies at the bottom for revenue are slightly off but still close to being accurate.

In [10]:
data.describe()

Rank         Year  Runtime (Minutes)       Rating         Votes  \
count  1000.000000  1000.000000        1000.000000  1000.000000  1.000000e+03   
mean    500.500000  2012.783000         113.172000     6.723200  1.698083e+05   
std     288.819436     3.205962          18.810908     0.945429  1.887626e+05   
min       1.000000  2006.000000          66.000000     1.900000  6.100000e+01   
25%     250.750000  2010.000000         100.000000     6.200000  3.630900e+04   
50%     500.500000  2014.000000         111.000000     6.800000  1.107990e+05   
75%     750.250000  2016.000000         123.000000     7.400000  2.399098e+05   
max    1000.000000  2016.000000         191.000000     9.000000  1.791916e+06   

       Revenue (Millions)   Metascore  
count          872.000000  936.000000  
mean            82.956376   58.985043  
std            103.253540   17.194757  
min              0.000000   11.000000  
25%             13.270000   47.000000  
50%             47.985000   59.500000  
75%            113.715000   72.000000  
max            936.630000  100.000000

As an example, what we can do with our data is only print out the genre of the 1,000 movies, as seen below.

In [18]:
genre = data['Genre']

In [19]:
data[['Genre']]

Genre
0     Action,Adventure,Sci-Fi
1    Adventure,Mystery,Sci-Fi
2             Horror,Thriller
3     Animation,Comedy,Family
4    Action,Adventure,Fantasy
..                        ...
995       Crime,Drama,Mystery
996                    Horror
997       Drama,Music,Romance
998          Adventure,Comedy
999     Comedy,Family,Fantasy

[1000 rows x 1 columns]

Now for a more introspective look at our data, we can print out specific columns that we want. We're printing out the title, genre, the actors, director, and rating of a movie. What we can do though, is type in whatever movie we want and get our results. Below, **Guardians of the Galaxy** was typed in as an example, but we could use any movie from the list that we want.

In [20]:
some_cols = data[['Title','Genre','Actors','Director','Rating']]

In [23]:
data_indexed.loc[['Guardians of the Galaxy']][['Genre','Actors','Director','Rating','Revenue (Millions)']]

Genre  \
Title                                              
Guardians of the Galaxy  Action,Adventure,Sci-Fi   

                                                                    Actors  \
Title                                                                        
Guardians of the Galaxy  Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...   

                           Director  Rating  Revenue (Millions)  
Title                                                            
Guardians of the Galaxy  James Gunn     8.1              333.13

We can also get specific with our data like below. With our code here, we're specifying movies that came out between 2014 and 2016, but we can change this to whatever year we want that's in the range of 2006-2016 since that's the range of our data. What we're also doing beyond specifying the years for movies in which they came out, is that we're also specifying that the movie has to have a rating of below six along with revenue being in the 0.80 quantile. So the movie had to do fairly well for itself at the box office. Again, these are the revenue numbers domestically, not internationally or worldwide. 

In [27]:
data[((data['Year'] >= 2014) & (data['Year'] <= 2016))
      & (data['Rating'] < 6.0)
      & (data['Revenue (Millions)'] > data['Revenue (Millions)'].quantile(0.80))]

Rank                            Title                    Genre  \
63     64             Fifty Shades of Grey   Drama,Romance,Thriller   
126   127  Transformers: Age of Extinction  Action,Adventure,Sci-Fi   
657   658     Teenage Mutant Ninja Turtles  Action,Adventure,Comedy   

                                           Description            Director  \
63   Literature student Anastasia Steele's life cha...  Sam Taylor-Johnson   
126  Autobots must escape sight from a bounty hunte...         Michael Bay   
657  When a kingpin threatens New York City, a grou...  Jonathan Liebesman   

                                                Actors  Year  \
63   Dakota Johnson, Jamie Dornan, Jennifer Ehle,El...  2015   
126  Mark Wahlberg, Nicola Peltz, Jack Reynor, Stan...  2014   
657  Megan Fox, Will Arnett, William Fichtner, Noel...  2014   

     Runtime (Minutes)  Rating   Votes  Revenue (Millions)  Metascore  
63                 125     4.1  244474              166.15       46.0  
126                165     5.7  255483              245.43       32.0  
657                101     5.9  178527              190.87       31.0

We can also group our data, such as here, with the director and their average rating for the movies they've directed.

In [28]:
data.groupby('Director')[['Rating']].mean().head()

Rating
Director                   
Aamir Khan              8.5
Abdellatif Kechiche     7.8
Adam Leon               6.5
Adam McKay              7.0
Adam Shankman           6.3

Below is an example of this idea where we grouped directors and their average ratings for their movies. The difference with this column compared to the one above is that we put the list from greatest rating to least. So, this list is the directors with the top ten highest average movie ratings that they've directed. This is done with the code **ascending=False** at the end of the code. The top director, as we can see, is Nitesh Tiwari.

In [63]:
data.groupby('Director')[['Rating']].mean().sort_values(['Rating'], ascending=False).head(10)

Rating
Director                                
Nitesh Tiwari                       8.80
Christopher Nolan                   8.68
Olivier Nakache                     8.60
Makoto Shinkai                      8.60
Aamir Khan                          8.50
Florian Henckel von Donnersmarck    8.50
Naoko Yamada                        8.40
Damien Chazelle                     8.40
Lee Unkrich                         8.30
Amber Tamblyn                       8.30

On the flip side, we can see who the bottom ten directors are when it comes to average rating by changing **ascending=False** to **ascending=True**. The director, as we can see with the lowest average rating, is Jason Friedberg.

In [64]:
data.groupby('Director')[['Rating']].mean().sort_values(['Rating'], ascending=True).head(10)

Rating
Director                  
Jason Friedberg        1.9
James Wong             2.7
Shawn Burkett          2.7
Jonathan Holbrook      3.2
Femi Oyeniran          3.5
Micheal Bafaro         3.5
Jeffrey G. Hunt        3.7
Rolfe Kanefsky         3.9
Joey Curtis            4.0
Sam Taylor-Johnson     4.1

When looking through the Excel file and also pulling the null data from the code below, it's clear there are rows where there's nothing input for the revenue or Metascore. 

In [33]:
data.isnull().sum()

Rank                    0
Title                   0
Genre                   0
Description             0
Director                0
Actors                  0
Year                    0
Runtime (Minutes)       0
Rating                  0
Votes                   0
Revenue (Millions)    128
Metascore              64
dtype: int64

What we can do is drop the Metascore column completely from the data. **axis=1** is saying that the column is to be dropped. These changes don't take place unless we specify **inplace=True** as a parameter in the **drop()** function.

As we can see below, there's no Metascore column anymore.

In [34]:
data.drop('Metascore', axis=1).head()

Rank                    Title                     Genre  \
0     1  Guardians of the Galaxy   Action,Adventure,Sci-Fi   
1     2               Prometheus  Adventure,Mystery,Sci-Fi   
2     3                    Split           Horror,Thriller   
3     4                     Sing   Animation,Comedy,Family   
4     5            Suicide Squad  Action,Adventure,Fantasy   

                                         Description              Director  \
0  A group of intergalactic criminals are forced ...            James Gunn   
1  Following clues to the origin of mankind, a te...          Ridley Scott   
2  Three girls are kidnapped by a man with a diag...    M. Night Shyamalan   
3  In a city of humanoid animals, a hustling thea...  Christophe Lourdelet   
4  A secret government agency recruits some of th...            David Ayer   

                                              Actors  Year  Runtime (Minutes)  \
0  Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014                121   
1  Noomi Rapace, Logan Marshall-Green, Michael Fa...  2012                124   
2  James McAvoy, Anya Taylor-Joy, Haley Lu Richar...  2016                117   
3  Matthew McConaughey,Reese Witherspoon, Seth Ma...  2016                108   
4  Will Smith, Jared Leto, Margot Robbie, Viola D...  2016                123   

   Rating   Votes  Revenue (Millions)  
0     8.1  757074              333.13  
1     7.0  485820              126.46  
2     7.3  157606              138.12  
3     7.2   60545              270.32  
4     6.2  393727              325.02

Using **data.dropna()** is important because this is where we can drop those null values that we discussed earlier that can mess with the data. These null values, if you remember, were in the **Revenue** and **Metascore** columns.

In [35]:
data.dropna()

Rank                     Title                     Genre  \
0       1   Guardians of the Galaxy   Action,Adventure,Sci-Fi   
1       2                Prometheus  Adventure,Mystery,Sci-Fi   
2       3                     Split           Horror,Thriller   
3       4                      Sing   Animation,Comedy,Family   
4       5             Suicide Squad  Action,Adventure,Fantasy   
..    ...                       ...                       ...   
993   994  Resident Evil: Afterlife   Action,Adventure,Horror   
994   995                 Project X                    Comedy   
996   997           Hostel: Part II                    Horror   
997   998    Step Up 2: The Streets       Drama,Music,Romance   
999  1000                Nine Lives     Comedy,Family,Fantasy   

                                           Description              Director  \
0    A group of intergalactic criminals are forced ...            James Gunn   
1    Following clues to the origin of mankind, a te...          Ridley Scott   
2    Three girls are kidnapped by a man with a diag...    M. Night Shyamalan   
3    In a city of humanoid animals, a hustling thea...  Christophe Lourdelet   
4    A secret government agency recruits some of th...            David Ayer   
..                                                 ...                   ...   
993  While still out to destroy the evil Umbrella C...    Paul W.S. Anderson   
994  3 high school seniors throw a birthday party t...       Nima Nourizadeh   
996  Three American college students studying abroa...              Eli Roth   
997  Romantic sparks occur between two dance studen...            Jon M. Chu   
999  A stuffy businessman finds himself trapped ins...      Barry Sonnenfeld   

                                                Actors  Year  \
0    Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014   
1    Noomi Rapace, Logan Marshall-Green, Michael Fa...  2012   
2    James McAvoy, Anya Taylor-Joy, Haley Lu Richar...  2016   
3    Matthew McConaughey,Reese Witherspoon, Seth Ma...  2016   
4    Will Smith, Jared Leto, Margot Robbie, Viola D...  2016   
..                                                 ...   ...   
993  Milla Jovovich, Ali Larter, Wentworth Miller,K...  2010   
994  Thomas Mann, Oliver Cooper, Jonathan Daniel Br...  2012   
996  Lauren German, Heather Matarazzo, Bijou Philli...  2007   
997  Robert Hoffman, Briana Evigan, Cassie Ventura,...  2008   
999  Kevin Spacey, Jennifer Garner, Robbie Amell,Ch...  2016   

     Runtime (Minutes)  Rating   Votes  Revenue (Millions)  Metascore  
0                  121     8.1  757074              333.13       76.0  
1                  124     7.0  485820              126.46       65.0  
2                  117     7.3  157606              138.12       62.0  
3                  108     7.2   60545              270.32       59.0  
4                  123     6.2  393727              325.02       40.0  
..                 ...     ...     ...                 ...        ...  
993                 97     5.9  140900               60.13       37.0  
994                 88     6.7  164088               54.72       48.0  
996                 94     5.5   73152               17.54       46.0  
997                 98     6.2   70699               58.01       50.0  
999                 87     5.3   12435               19.64       11.0  

[838 rows x 12 columns]

The code below is what drops all of the columns containing missing data. You will see there are no revenue or Metascore columns.

In [36]:
data.dropna(axis=1)

Rank                    Title                     Genre  \
0       1  Guardians of the Galaxy   Action,Adventure,Sci-Fi   
1       2               Prometheus  Adventure,Mystery,Sci-Fi   
2       3                    Split           Horror,Thriller   
3       4                     Sing   Animation,Comedy,Family   
4       5            Suicide Squad  Action,Adventure,Fantasy   
..    ...                      ...                       ...   
995   996     Secret in Their Eyes       Crime,Drama,Mystery   
996   997          Hostel: Part II                    Horror   
997   998   Step Up 2: The Streets       Drama,Music,Romance   
998   999             Search Party          Adventure,Comedy   
999  1000               Nine Lives     Comedy,Family,Fantasy   

                                           Description              Director  \
0    A group of intergalactic criminals are forced ...            James Gunn   
1    Following clues to the origin of mankind, a te...          Ridley Scott   
2    Three girls are kidnapped by a man with a diag...    M. Night Shyamalan   
3    In a city of humanoid animals, a hustling thea...  Christophe Lourdelet   
4    A secret government agency recruits some of th...            David Ayer   
..                                                 ...                   ...   
995  A tight-knit team of rising investigators, alo...             Billy Ray   
996  Three American college students studying abroa...              Eli Roth   
997  Romantic sparks occur between two dance studen...            Jon M. Chu   
998  A pair of friends embark on a mission to reuni...        Scot Armstrong   
999  A stuffy businessman finds himself trapped ins...      Barry Sonnenfeld   

                                                Actors  Year  \
0    Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014   
1    Noomi Rapace, Logan Marshall-Green, Michael Fa...  2012   
2    James McAvoy, Anya Taylor-Joy, Haley Lu Richar...  2016   
3    Matthew McConaughey,Reese Witherspoon, Seth Ma...  2016   
4    Will Smith, Jared Leto, Margot Robbie, Viola D...  2016   
..                                                 ...   ...   
995  Chiwetel Ejiofor, Nicole Kidman, Julia Roberts...  2015   
996  Lauren German, Heather Matarazzo, Bijou Philli...  2007   
997  Robert Hoffman, Briana Evigan, Cassie Ventura,...  2008   
998  Adam Pally, T.J. Miller, Thomas Middleditch,Sh...  2014   
999  Kevin Spacey, Jennifer Garner, Robbie Amell,Ch...  2016   

     Runtime (Minutes)  Rating   Votes  
0                  121     8.1  757074  
1                  124     7.0  485820  
2                  117     7.3  157606  
3                  108     7.2   60545  
4                  123     6.2  393727  
..                 ...     ...     ...  
995                111     6.2   27585  
996                 94     5.5   73152  
997                 98     6.2   70699  
998                 93     5.6    4881  
999                 87     5.3   12435  

[1000 rows x 10 columns]

Next, we're using the thresh parameter to specify the minimum number of non-null values for the column/row to be held without dropping. This is to clean up the data, so we don't have a bunch of null values present in our data.

In [37]:
data.dropna(axis=0, thresh=6)

Rank                    Title                     Genre  \
0       1  Guardians of the Galaxy   Action,Adventure,Sci-Fi   
1       2               Prometheus  Adventure,Mystery,Sci-Fi   
2       3                    Split           Horror,Thriller   
3       4                     Sing   Animation,Comedy,Family   
4       5            Suicide Squad  Action,Adventure,Fantasy   
..    ...                      ...                       ...   
995   996     Secret in Their Eyes       Crime,Drama,Mystery   
996   997          Hostel: Part II                    Horror   
997   998   Step Up 2: The Streets       Drama,Music,Romance   
998   999             Search Party          Adventure,Comedy   
999  1000               Nine Lives     Comedy,Family,Fantasy   

                                           Description              Director  \
0    A group of intergalactic criminals are forced ...            James Gunn   
1    Following clues to the origin of mankind, a te...          Ridley Scott   
2    Three girls are kidnapped by a man with a diag...    M. Night Shyamalan   
3    In a city of humanoid animals, a hustling thea...  Christophe Lourdelet   
4    A secret government agency recruits some of th...            David Ayer   
..                                                 ...                   ...   
995  A tight-knit team of rising investigators, alo...             Billy Ray   
996  Three American college students studying abroa...              Eli Roth   
997  Romantic sparks occur between two dance studen...            Jon M. Chu   
998  A pair of friends embark on a mission to reuni...        Scot Armstrong   
999  A stuffy businessman finds himself trapped ins...      Barry Sonnenfeld   

                                                Actors  Year  \
0    Chris Pratt, Vin Diesel, Bradley Cooper, Zoe S...  2014   
1    Noomi Rapace, Logan Marshall-Green, Michael Fa...  2012   
2    James McAvoy, Anya Taylor-Joy, Haley Lu Richar...  2016   
3    Matthew McConaughey,Reese Witherspoon, Seth Ma...  2016   
4    Will Smith, Jared Leto, Margot Robbie, Viola D...  2016   
..                                                 ...   ...   
995  Chiwetel Ejiofor, Nicole Kidman, Julia Roberts...  2015   
996  Lauren German, Heather Matarazzo, Bijou Philli...  2007   
997  Robert Hoffman, Briana Evigan, Cassie Ventura,...  2008   
998  Adam Pally, T.J. Miller, Thomas Middleditch,Sh...  2014   
999  Kevin Spacey, Jennifer Garner, Robbie Amell,Ch...  2016   

     Runtime (Minutes)  Rating   Votes  Revenue (Millions)  Metascore  
0                  121     8.1  757074              333.13       76.0  
1                  124     7.0  485820              126.46       65.0  
2                  117     7.3  157606              138.12       62.0  
3                  108     7.2   60545              270.32       59.0  
4                  123     6.2  393727              325.02       40.0  
..                 ...     ...     ...                 ...        ...  
995                111     6.2   27585                 NaN       45.0  
996                 94     5.5   73152               17.54       46.0  
997                 98     6.2   70699               58.01       50.0  
998                 93     5.6    4881                 NaN       22.0  
999                 87     5.3   12435               19.64       11.0  

[1000 rows x 12 columns]

Now we're finding the mean in revenue for movies that have a revenue value in their column. With this, we see that the mean revenue with movies that have a revenue value tied to them is roughly 83 million. 

In [38]:
revenue_mean = data_indexed['Revenue (Millions)'].mean()
print("The mean revenue is: ", revenue_mean)

The mean revenue is:  82.95637614678897


Now with the code below, we will use the mean revenue we just calculated to input into these null revenue values to fill in these null values. So when the data frame is checked, there won't be any null values in the revenue column. The reason for using the mean revenue to fill in null values is that it gives us a closely accurate picture of the total mean revenue if these null values had a value present. So when we calculate the mean of all 1,000 movies, we get a more accurate picture since the null values aren't weighing the true mean value of the movies down.

In [39]:
data_indexed['Revenue (Millions)'].fillna(revenue_mean, inplace=True)

We can use the **apply()** function, which allows us to get technical in terms of how we group the ratings of our movies along with labeling whether the movie was good, average, or bad based on the rating. So with the code below, we created a **rating_group** function that allows us to do this. As we can see, if the rating is greater than or equal to 7.5, the movie would be considered **Good**. If it's equal to or greater than 6 and less than 7.5, the movie will be considered **Average**. If the rating is below 6, the movie will be considered **Bad**.

In [40]:
def rating_group(rating):
    if rating >= 7.5:
        return 'Good'
    elif rating >= 6.0:
        return 'Average'
    else:
        return 'Bad'

This is where we apply the **rating_group** function

In [41]:
data['Rating_category'] = data['Rating'].apply(rating_group)

In [71]:
data[['Title','Director','Rating','Rating_category']].head(30)


Title              Director  Rating  \
0                   Guardians of the Galaxy            James Gunn     8.1   
1                                Prometheus          Ridley Scott     7.0   
2                                     Split    M. Night Shyamalan     7.3   
3                                      Sing  Christophe Lourdelet     7.2   
4                             Suicide Squad            David Ayer     6.2   
5                            The Great Wall           Yimou Zhang     6.1   
6                                La La Land       Damien Chazelle     8.3   
7                                  Mindhorn            Sean Foley     6.4   
8                        The Lost City of Z            James Gray     7.1   
9                                Passengers         Morten Tyldum     7.0   
10  Fantastic Beasts and Where to Find Them           David Yates     7.5   
11                           Hidden Figures        Theodore Melfi     7.8   
12                                Rogue One        Gareth Edwards     7.9   
13                                    Moana          Ron Clements     7.7   
14                                 Colossal       Nacho Vigalondo     6.4   
15                  The Secret Life of Pets          Chris Renaud     6.6   
16                            Hacksaw Ridge            Mel Gibson     8.2   
17                             Jason Bourne       Paul Greengrass     6.7   
18                                     Lion           Garth Davis     8.1   
19                                  Arrival      Denis Villeneuve     8.0   
20                                     Gold        Stephen Gaghan     6.7   
21                    Manchester by the Sea      Kenneth Lonergan     7.9   
22                           Hounds of Love             Ben Young     6.7   
23                                   Trolls            Walt Dohrn     6.5   
24             Independence Day: Resurgence       Roland Emmerich     5.3   
25                          Paris pieds nus        Dominique Abel     6.8   
26                  Bahubali: The Beginning        S.S. Rajamouli     8.3   
27                               Dead Awake        Phillip Guzman     4.7   
28                                 Bad Moms             Jon Lucas     6.2   
29                         Assassin's Creed         Justin Kurzel     5.9   

   Rating_category  
0             Good  
1          Average  
2          Average  
3          Average  
4          Average  
5          Average  
6             Good  
7          Average  
8          Average  
9          Average  
10            Good  
11            Good  
12            Good  
13            Good  
14         Average  
15         Average  
16            Good  
17         Average  
18            Good  
19            Good  
20         Average  
21            Good  
22         Average  
23         Average  
24             Bad  
25         Average  
26            Good  
27             Bad  
28         Average  
29             Bad

In [ ]:
data.groupby('Director')[['Rating']].mean().sort_values(['Rating'], ascending=True).head()

As we can see from above, we have the title of the movie, the director, rating, and the **rating_group** function we created has been renamed to **Rating_category** where we see the function being applied. I decided to print the first thirty results from the list with the **head()** function so that we could see the **Good**, **Average**, and **Bad** ratings on display.

!["C:\Users\14dav\miniconda3\Lib\site-packages\nbclassic\static\components\requirejs-plugins\src\imdb.png"](imdb.png)

Here, I have organized the highest-grossing movies from the list regarding revenue and looked for a correlation. What we can see from this chart is that movies with the highest domestic revenue don't have a common director that's dominating the list, but they have a common genre. Almost all of these movies have the same couple of related genres: **Action**, **Adventure**, and **Sci-Fi**. You could also sprinkle in **Fantasy** and **Comedy**, but the other three are the main staples. 
So, when looking at the top 1,000 IMDb movies from 2006-2016, we can see a correlation between the movies with the highest revenue and their genre. There's also some fascinating insight into movies with low Metascore ratings that still have a high revenue. For example, when we printed out movies with a rating below six and were at the 0.80 quantile in revenue, we saw movies such as **Fifty Shades of Grey** bringing in 166.15 million dollars even with a 4.1 rating. We also had **Transformers: Age of Extinction** with a rating of 5.7 that brought in 245.43 million dollars. So, just because a movie has a low rating doesn't mean there's no potential to make a lot of money at the box office.